037-WindowsVirtualDesktop

jamasten opened the pull request:
adding hack for WVD

jrzyshr reviewed (changes requested):
Hi Team!  Thanks for the great new entry to WTH!

Thanks for fixing all of the navigation links as per our in-person review.

There are changes I'm going to request for the /Student/Resources folder.  I've shared those internally on Teams with Jason Masten.  I'll wait for you to take a stab at them before we review & publish this PR.

gfilicetti commented on 037-WindowsVirtualDesktop/README.md:
```diff
@@ -0,0 +1,55 @@
+- Azure Subscription
+- M365 License
+- Azure experience
+- [Azure Academy Az-140 Study Guide](https://aka.ms/AzureAcademy-AZ140)
```
Is AZ140 truly a prereq? Deal breaker if you dont' do it? It's hard to get students to do stuff pre-event so consider putting "(Optional)" at the end of this one.

But I will go with whatever you decide, just something to think about.

gfilicetti commented on 037-WindowsVirtualDesktop/README.md:
```diff
@@ -0,0 +1,55 @@
+- Challenge 8: **[Plan and Implement BCDR for your WVD Environment](Student/08-Plan-Implement-BCDR.md)**
+- Challenge 9: **[Automation of your WVD Tasks](Student/09-Automate-WVD-Tasks.md)**
+- Challenge 10: **[Monitor and Manage the performance health of your environment](Student/10-Monitor-Manage-Performance-Health.md)**
+- Challenge 11: **[Configure settings for user experience](Student/11-Configure-User-Experience-Settings.md)**
```
All of the link texts for this table of contents should match the title header in the challenge text itself. Right now there are some mismatches. Up to you which of the two wins, but they should match.

Also I would capitalize every word in a section header/title except the smallest words, 'a', 'the', 'of'

gfilicetti commented on 037-WindowsVirtualDesktop/Student/00-Pre-Reqs.md:
```diff
@@ -0,0 +1,34 @@
+# Challenge 0: Getting started
```
This needs to match the table of contents.... same applies for all challenges so I'll only make this comment once.

gfilicetti commented on 037-WindowsVirtualDesktop/Student/01-Plan-WVD-Architecture.md:
```diff
@@ -0,0 +1,60 @@
+
+## Success Criteria
+
+This Challenge will be succesfully completed when you have created a document defining all of the parameters that will be used in the deployment challenges.  
```
I would change this from a sentence to a bullet list, even if it is a list of 1, such as:

- You have created a document defining all parameters used in the deployment challenges.

gfilicetti commented on 037-WindowsVirtualDesktop/Student/01-Plan-WVD-Architecture.md:
```diff
@@ -0,0 +1,60 @@
+
+[Role based access control](https://docs.microsoft.com/en-us/azure/virtual-desktop/rbac)
+
+## Tips
```
Please move Tips section about Learning Resources. Learning Resources should be last on the page, always.

gfilicetti commented on 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md:
```diff
@@ -0,0 +1,67 @@
+            * **NOTE:**: Only install FSLogix. The configuration will occur in a future challenge.
+        * Notepad++
+        * Adobe Acrobat Reader
+    * **HINT:**: Since this operating system supports more than one user logged in at a time, there may be specific installation steps to consider.
```
You have a double semicolon after HINT

gfilicetti commented on 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md:
```diff
@@ -0,0 +1,67 @@
+    * Create an Image Template in AIB with the following customizations:
+        * The operating system is Windows 10 Enterprise Multi-session, Version 20H2 + Microsoft 365 Apps (Gen 1) with the following software installed.
+            * Microsoft Teams
+            * **HINT:**: Since this operating system is multi-session, there may be specific installation steps to consider.
```
Another double semicolon after HINT.

gfilicetti commented on 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md:
```diff
@@ -0,0 +1,67 @@
+        * Microsoft OneDrive
+        * Microsoft Teams
+        * FSLogix
+            * **NOTE:**: Only install FSLogix. The configuration will occur in a future challenge.
```
You have a double semi colon after NOTE

gfilicetti commented on 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md:
```diff
@@ -0,0 +1,67 @@
+            * **HINT:**: Since this operating system is multi-session, there may be specific installation steps to consider.
+        * Add a step to reboot the VM.
+        * Add a step to run Windows Update.
+        * **NOTE** All Windows 10 Multi-session images from the Azure Marketplace already include the FSLogix agent.
```
Missing a semicolon after NOTE

gfilicetti commented on 037-WindowsVirtualDesktop/Student/05-Create-Configure-HostPools.md:
```diff
@@ -0,0 +1,45 @@
+
+## Introduction
+
+In this challenge you will Create and Configure Host Pools and Session Hosts for your environment. 
```
Unnecessary capitalization here, this is a sentence, not a section header, only the first letter needs capitalization.

gfilicetti commented on 037-WindowsVirtualDesktop/Student/05-Create-Configure-HostPools.md:
```diff
@@ -0,0 +1,45 @@
+In this challenge we will be provisioning our Host Pools and Session Hosts. You will need to create four Host Pools with three different OS. You will also need to use three different ways to deploy those HostPools.
+Once you created the Host Pools you will need to add Session Hosts, assign users to the Host Pools, and configure Host Pool settings. VMs for each region should be added to the domain's region OU. 
+
+**Note:** ALL HOST POOLS MUST BE CONFIGURED AS VALIDATION POOLS
```
Note should be: NOTE

gfilicetti commented on 037-WindowsVirtualDesktop/Student/06-Implement-Manage-FsLogix.md:
```diff
@@ -0,0 +1,30 @@
+2. Solution should allow users to access profile data if a region is unavailable
+3. Configure FSLogix with GPO
+
+**TIPS:** Teams cache data can be large sometimes (4-5 GB) and customers may want it omitted it from the profile. User FSLogix redirections so TEAMS data is kept on the local system, not in the users profile. 
```
Please make Tips it's own section like you have in Challenge 1

gfilicetti commented on 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md:
```diff
@@ -0,0 +1,47 @@
+
+**NOTE:** Scaling Automation was deployed in Challenge 9
+
+#### WVD Monitoring
```
I would make these subsections under 'Description' 3 level headers, not 4 level headers because Description is a 2 level header.

gfilicetti commented on 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md:
```diff
@@ -0,0 +1,47 @@
+- Identify performance metrics of a WVD host which has a currently connected user and demonstrate how you can view the "User Input Delay per Process" object and "Max Input Delay" counters for that VM.
+- Enable automatic scaling of WVD session hosts to start and stop machines during peak/off-peak hours across each region. Consider how the peak/off-peak hours and load balancing will influence the configuration. Assume for the purpose of this workshop that VM sizing and user density is designed for a 1CPU per Session to help with testing in a limited size environment.
+
+**Tips**
```
Tips needs to be a level 2 header

gfilicetti commented on 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md:
```diff
@@ -0,0 +1,47 @@
+> _Simulate an unavailable session host by manually stopping the WVD service "RdAgent"_
+
+
+Advanced Challenges (Optional)
```
Advanced Challenges needs to be a level 2 header

gfilicetti commented on 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md:
```diff
@@ -0,0 +1,47 @@
+
+> _To simulate connection errors you may wish to attempt logins with invalid credentials, closing the client during the logon attempt and deliberately using a different account between the logon to the WVD portal and the WVD session host which should fail._
+
+> _Simulate an unavailable session host by manually stopping the WVD service "RdAgent"_
```
Please don't use angle brackets on this text. We don't want it to appear as a quote on the rendered page.

gfilicetti commented on 037-WindowsVirtualDesktop/Coach/README.md:
```diff
@@ -0,0 +1,20 @@
+- Challenge 8: **[Plan and Implement BCDR for your WVD Environment](08-Plan-Implement-BCDR.md)**
+- Challenge 9: **[Automation of your WVD Tasks](09-Automate-WVD-Tasks.md)**
+- Challenge 10: **[Monitor and Manage the performance health of your environment](10-Monitor-Manage-Performance-Health.md)**
+- Challenge 11: **[Configure settings for user experience](11-Configure-User-Experience-Settings.md)**
```
Make sure this ToC text matches the headers in the Coach guide for each challenge and that same text should be used in the Student Guide.

gfilicetti commented on 037-WindowsVirtualDesktop/Coach/00-Pre-Reqs.md:
```diff
@@ -0,0 +1,20 @@
+- Quotas are set per region.  If you increase the quota in a single region, you need to ensure that all students deploy to the same region.  Or else, they will bump up against the quota limits in the region they deploy to.
+- If there is no access to an administrator who can request quota increases, have the students deploy to different regions to stay within the quota limit of each region.  
+
+**NOTE** Remember to point students to learning resources
```
Add a semicolon after NOTE

gfilicetti commented on 037-WindowsVirtualDesktop/Coach/03-Implement-Manage-Storage.md:
```diff
@@ -0,0 +1,53 @@
+
+## Notes & Guidance
+
+1. Capacity planning
```
I would use a bullet here, not a number. You don't have a #2 so it's not really an ordered list.

gfilicetti commented on 037-WindowsVirtualDesktop/Coach/10-Monitor-Manage-Performance-Health.md:
```diff
@@ -0,0 +1,141 @@
+
+    - Save work by first hitting the done editing button and choosing the "Save As" option before hitting Done edit. Users, if successful will have a new metric counter that they added.
+
+## Reference
```
Change this to 'Learning Resources', doesn't matter that it's in the coach guide.

gfilicetti reviewed (changes requested):
Great work guys... sorry to get all super nitpicky but that's just the way we are, should be fast and easy fixes.

jamasten commented:
Gino & Peter,

All requested changes have been fixed.  Please let us know if there is anything else that needs to be fixed.  Thank you!

Best,
Jason Masten
Customer Engineer | FastTrack for Azure
Azure Customer Experience Engineering (CXP)
Office Hours: 8am - 5pm Eastern Time Zone, USA
***@***.***D73702.BC243230]
Learn more about FastTrack for Azure, here<https://azure.microsoft.com/en-us/programs/azure-fasttrack/>.
We're HIRING: https://aka.ms/azurecxpjobs

This email message may contain confidential and proprietary information.  Any unauthorized use is prohibited.  If you are not the intended recipient, please contact the sender by reply email and destroy all copies of the original message.


From: Gino Filicetti ***@***.***>
Sent: Wednesday, April 21, 2021 9:58 PM
To: microsoft/WhatTheHack ***@***.***>
Cc: Jason Masten ***@***.***>; Author ***@***.***>
Subject: Re: [microsoft/WhatTheHack] 037-WindowsVirtualDesktop (#223)


@gfilicetti requested changes on this pull request.

Great work guys... sorry to get all super nitpicky but that's just the way we are, should be fast and easy fixes.

________________________________

In 037-WindowsVirtualDesktop/README.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618005852&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764592053%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=wq%2BuatpuBgamb5IE2igWOxTDczms6yy9%2FOgpaBolWUs%3D&reserved=0>:

> +- Challenge 3: **[Implement and Manage your storage for FsLogix](Student/03-Implement-Manage-Storage.md)**

+- Challenge 4: **[Create and Manage your Images for your Session Hosts](Student/04-Create-Manage-Images.md)**

+- Challenge 5: **[Create and Configure your WVD HostPools](Student/05-Create-Configure-HostPools.md)**

+- Challenge 6: **[Implement and Manage your FsLogix](Student/06-Implement-Manage-FsLogix.md)**

+- Challenge 7: **[Install and Configure your applications](Student/07-Install-Configure-Apps.md)**

+- Challenge 8: **[Plan and Implement BCDR for your WVD Environment](Student/08-Plan-Implement-BCDR.md)**

+- Challenge 9: **[Automation of your WVD Tasks](Student/09-Automate-WVD-Tasks.md)**

+- Challenge 10: **[Monitor and Manage the performance health of your environment](Student/10-Monitor-Manage-Performance-Health.md)**

+- Challenge 11: **[Configure settings for user experience](Student/11-Configure-User-Experience-Settings.md)**

+

+## Prerequisites

+

+- Azure Subscription

+- M365 License

+- Azure experience

+- [Azure Academy Az-140 Study Guide](https://aka.ms/AzureAcademy-AZ140)

Is AZ140 truly a prereq? Deal breaker if you dont' do it? It's hard to get students to do stuff pre-event so consider putting "(Optional)" at the end of this one.

But I will go with whatever you decide, just something to think about.

________________________________

In 037-WindowsVirtualDesktop/README.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618007281&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764592053%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=IMK2jZQyqh16rZLCHwqg28%2BA2lCGf3S0DkMIohTHF38%3D&reserved=0>:

> +In this hack you will learn how to set up a Windows Virtual Desktop environment from start to finish. You will take a customer scenario and build it out in your own environment. Once your WVD environment is built you will learn how to scale. monitor and manage the environment with Azure resources.

+

+## Challenges

+

+- Challenge 0: **[Windows Virtual Desktop  Pre-Reqs](Student/00-Pre-Reqs.md)**

+- Challenge 1: **[Planning your WVD Architecture](Student/01-Plan-WVD-Architecture.md)**

+- Challenge 2: **[Implementing and Managing your WVD Network](Student/02-Implement-Manage-Network.md)**

+- Challenge 3: **[Implement and Manage your storage for FsLogix](Student/03-Implement-Manage-Storage.md)**

+- Challenge 4: **[Create and Manage your Images for your Session Hosts](Student/04-Create-Manage-Images.md)**

+- Challenge 5: **[Create and Configure your WVD HostPools](Student/05-Create-Configure-HostPools.md)**

+- Challenge 6: **[Implement and Manage your FsLogix](Student/06-Implement-Manage-FsLogix.md)**

+- Challenge 7: **[Install and Configure your applications](Student/07-Install-Configure-Apps.md)**

+- Challenge 8: **[Plan and Implement BCDR for your WVD Environment](Student/08-Plan-Implement-BCDR.md)**

+- Challenge 9: **[Automation of your WVD Tasks](Student/09-Automate-WVD-Tasks.md)**

+- Challenge 10: **[Monitor and Manage the performance health of your environment](Student/10-Monitor-Manage-Performance-Health.md)**

+- Challenge 11: **[Configure settings for user experience](Student/11-Configure-User-Experience-Settings.md)**

All of the link texts for this table of contents should match the title header in the challenge text itself. Right now there are some mismatches. Up to you which of the two wins, but they should match.

Also I would capitalize every word in a section header/title except the smallest words, 'a', 'the', 'of'

________________________________

In 037-WindowsVirtualDesktop/Student/00-Pre-Reqs.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618007538&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764602048%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=S4urtS9yXb04QKle6GBz258hKhXW2M45d89PYFzE90I%3D&reserved=0>:

> @@ -0,0 +1,34 @@

+# Challenge 0: Getting started

This needs to match the table of contents.... same applies for all challenges so I'll only make this comment once.

________________________________

In 037-WindowsVirtualDesktop/Student/01-Plan-WVD-Architecture.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618008746&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764602048%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=PnpD0xjeVxITtc01go6bo2tRVFwkR7TAYiWkH9rc1U0%3D&reserved=0>:

> +## Description

+

+Taking into account our scenario, specify the details for the following:

+

+1. Host pool(s)

+1. Images

+1. Storage

+1. Clients

+1. Security

+1. Networking

+1. FSLogix

+

+

+## Success Criteria

+

+This Challenge will be succesfully completed when you have created a document defining all of the parameters that will be used in the deployment challenges.

I would change this from a sentence to a bullet list, even if it is a list of 1, such as:

  *   You have created a document defining all parameters used in the deployment challenges.

________________________________

In 037-WindowsVirtualDesktop/Student/01-Plan-WVD-Architecture.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618009218&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764612039%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=r0NtIj5x7286dDDXHl%2FFI0ukl80AWyt0j1fu%2FTPbvZI%3D&reserved=0>:

> +[Az-140 ep04 | Plan Your WVD Session Hosts](https://www.youtube.com/watch?v=HNCZ2pzr9mo&list=PL-V4YVm6AmwW1DBM25pwWYd1Lxs84ILZT&index=6)

+

+

+[WVD overview](https://docs.microsoft.com/en-us/azure/virtual-desktop/overview)

+

+[Virtual Machine Sizing](https://docs.microsoft.com/en-us/windows-server/remote/remote-desktop-services/virtual-machine-recs)

+

+[FSLogix](https://docs.microsoft.com/en-us/azure/architecture/example-scenario/wvd/windows-virtual-desktop-fslogix)

+

+[Remote Desktop Clients](https://docs.microsoft.com/en-us/windows-server/remote/remote-desktop-services/clients/remote-desktop-clients)

+

+[Security](https://docs.microsoft.com/en-us/azure/virtual-desktop/security-baseline)

+

+[Role based access control](https://docs.microsoft.com/en-us/azure/virtual-desktop/rbac)

+

+## Tips

Please move Tips section about Learning Resources. Learning Resources should be last on the page, always.

________________________________

In 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618010021&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764622032%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=b1XkhW8VcKb%2FlfKh%2F5Mz8O5x0mbNGDyjmdLmJcCVhk0%3D&reserved=0>:

> +

+## Description

+

+In this challenge you will be preparing three images that will be used for the different Host Pool deployments. Each image will be use a different operating system and installed applications based on the different user personas. Each image will be created and updated differently based on the scenario requirements.

+

+1. Create the image for the Remote Field users

+    * This image should be created with the Virtual Machine Capture command and the custom image should only be stored in UK South.

+    * Create a virtual machine in UK South with the following specifications. The operating system is Windows Server 2019 Datacenter with the following software installed.

+        * Microsoft Office

+        * Microsoft OneDrive

+        * Microsoft Teams

+        * FSLogix

+            * **NOTE:**: Only install FSLogix. The configuration will occur in a future challenge.

+        * Notepad++

+        * Adobe Acrobat Reader

+    * **HINT:**: Since this operating system supports more than one user logged in at a time, there may be specific installation steps to consider.

You have a double semicolon after HINT

________________________________

In 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618010204&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764622032%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=s8LomYAGOR3HvggiV4DrrR8%2FDSUwvcJ7kt6TdkXLjbg%3D&reserved=0>:

> +            * **NOTE:**: Only install FSLogix. The configuration will occur in a future challenge.

+        * Notepad++

+        * Adobe Acrobat Reader

+    * **HINT:**: Since this operating system supports more than one user logged in at a time, there may be specific installation steps to consider.

+    * **HINT:** It is recommended to reboot the server after making the customizations to the image.

+    * Take a disk snapshot as a recovery point for any subsequent failures and the snapshot will be used when updating the image.

+    * Run Sysprep (generalize). Then Capture the VM as a custom image.

+    * **TIP:** Since the Capture makes the VM no longer usable, it is recommended to let the Capture delete the original VM.

+    * Test the image by deploying a VM from the image and validating all customizations persisted.

+

+1. Create the Image for the General Office users

+    * This image should be created with the Azure Image Builder and the image should be stored in the Shared Image Gallery (SIG). The image in SIG should be available in Japan West, UK South, and US East with three image replicas per region.

+    * Create an Image Template in AIB with the following customizations:

+        * The operating system is Windows 10 Enterprise Multi-session, Version 20H2 + Microsoft 365 Apps (Gen 1) with the following software installed.

+            * Microsoft Teams

+            * **HINT:**: Since this operating system is multi-session, there may be specific installation steps to consider.

Another double semicolon after HINT.

________________________________

In 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618010506&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764632029%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2Ftnw%2FkcSYWFUFvaMQjfk0m69nCACQ280PR%2B03NE%2FnfA%3D&reserved=0>:

> +

+In this challenge you will create, update, and manage the images used when deploying Session Hosts into your Host Pools.

+

+

+## Description

+

+In this challenge you will be preparing three images that will be used for the different Host Pool deployments. Each image will be use a different operating system and installed applications based on the different user personas. Each image will be created and updated differently based on the scenario requirements.

+

+1. Create the image for the Remote Field users

+    * This image should be created with the Virtual Machine Capture command and the custom image should only be stored in UK South.

+    * Create a virtual machine in UK South with the following specifications. The operating system is Windows Server 2019 Datacenter with the following software installed.

+        * Microsoft Office

+        * Microsoft OneDrive

+        * Microsoft Teams

+        * FSLogix

+            * **NOTE:**: Only install FSLogix. The configuration will occur in a future challenge.

You have a double semi colon after NOTE

________________________________

In 037-WindowsVirtualDesktop/Student/04-Create-Manage-Images.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618011029&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764632029%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=d5uxoS8TPfIrWCEidbZbinWCq5%2F00rLyqUKtoLgIxCM%3D&reserved=0>:

> +    * **HINT:**: Since this operating system supports more than one user logged in at a time, there may be specific installation steps to consider.

+    * **HINT:** It is recommended to reboot the server after making the customizations to the image.

+    * Take a disk snapshot as a recovery point for any subsequent failures and the snapshot will be used when updating the image.

+    * Run Sysprep (generalize). Then Capture the VM as a custom image.

+    * **TIP:** Since the Capture makes the VM no longer usable, it is recommended to let the Capture delete the original VM.

+    * Test the image by deploying a VM from the image and validating all customizations persisted.

+

+1. Create the Image for the General Office users

+    * This image should be created with the Azure Image Builder and the image should be stored in the Shared Image Gallery (SIG). The image in SIG should be available in Japan West, UK South, and US East with three image replicas per region.

+    * Create an Image Template in AIB with the following customizations:

+        * The operating system is Windows 10 Enterprise Multi-session, Version 20H2 + Microsoft 365 Apps (Gen 1) with the following software installed.

+            * Microsoft Teams

+            * **HINT:**: Since this operating system is multi-session, there may be specific installation steps to consider.

+        * Add a step to reboot the VM.

+        * Add a step to run Windows Update.

+        * **NOTE** All Windows 10 Multi-session images from the Azure Marketplace already include the FSLogix agent.

Missing a semicolon after NOTE

________________________________

In 037-WindowsVirtualDesktop/Student/05-Create-Configure-HostPools.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618012005&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764642025%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=HOKCWy0LkjEF%2FyXp%2BvRCoWxQVmDsQ1YoweJmFk%2BOWCQ%3D&reserved=0>:

> @@ -0,0 +1,45 @@

+# Challenge 5: Create and configure host pools and session hosts

+

+[< Previous Challenge](./04-Create-Manage-Images.md) - **[Home](../README.md)** - [Next Challenge >](./06-Implement-Manage-FsLogix.md)

+

+## Introduction

+

+In this challenge you will Create and Configure Host Pools and Session Hosts for your environment.

Unnecessary capitalization here, this is a sentence, not a section header, only the first letter needs capitalization.

________________________________

In 037-WindowsVirtualDesktop/Student/05-Create-Configure-HostPools.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618012245&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764642025%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=NCMm7uzOxsMDXZ7C%2FFTPzhH6sOVhiNpX4vVzP5QUXTA%3D&reserved=0>:

> @@ -0,0 +1,45 @@

+# Challenge 5: Create and configure host pools and session hosts

+

+[< Previous Challenge](./04-Create-Manage-Images.md) - **[Home](../README.md)** - [Next Challenge >](./06-Implement-Manage-FsLogix.md)

+

+## Introduction

+

+In this challenge you will Create and Configure Host Pools and Session Hosts for your environment.

+

+## Description

+

+In this challenge we will be provisioning our Host Pools and Session Hosts. You will need to create four Host Pools with three different OS. You will also need to use three different ways to deploy those HostPools.

+Once you created the Host Pools you will need to add Session Hosts, assign users to the Host Pools, and configure Host Pool settings. VMs for each region should be added to the domain's region OU.

+

+**Note:** ALL HOST POOLS MUST BE CONFIGURED AS VALIDATION POOLS

Note should be: NOTE

________________________________

In 037-WindowsVirtualDesktop/Student/06-Implement-Manage-FsLogix.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618012998&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764652024%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=iXsgbi5Aup42avOHZMzXikS7aA2zk6XiYSFqk3pr%2BWk%3D&reserved=0>:

> +- Configure the FSLogix clients

+    - Japan Users do not use FSLogix because they have personal hostpools.

+    - Configure VHDLocations for US Users.

+    - Configure Cloud Cache for UK Users.

+    - Optimize - Configure for VHDx | Configure Dynamic Disk | Replace local profile | set VHDx default size | Flip Flop directory name]

+    - Use GPO Preference to add admins to the local FSLogix exclude groups

+    - Use GPO Preference to add The correct users to the local FSLogix include groups

+

+

+## Success Criteria

+

+1. Configure FSLogix according to best practices

+2. Solution should allow users to access profile data if a region is unavailable

+3. Configure FSLogix with GPO

+

+**TIPS:** Teams cache data can be large sometimes (4-5 GB) and customers may want it omitted it from the profile. User FSLogix redirections so TEAMS data is kept on the local system, not in the users profile.

Please make Tips it's own section like you have in Challenge 1

________________________________

In 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618017119&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764652024%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=PhJ3cQDgXKFplpCI%2B0Nn2z0udzdfEuhsZpKwifOkJTo%3D&reserved=0>:

> @@ -0,0 +1,47 @@

+# Challenge 10: Monitor and manage the performance and health of your WVD environment

+

+[< Previous Challenge](./09-Automate-WVD-Tasks.md) - **[Home](../README.md)** - [Next Challenge>](./11-Configure-User-Experience-Settings.md)

+

+## Introduction

+

+You've recently deployed WVD for your company and each region started off ok however users have begun reporting problems occasional issues with connecting and performance has become progressively worse for some of the sessions. You need to offer a way to gather information to troubleshoot these issues and because they happen infrequently and across a range of users and times you need to be able to efficiently capture and record these events. It's important that you also can be more proactive to issues in the environment before they cause too much impact on the user experience. Your manager is also concerned about the cost of running so many WVD hosts across each of the regions when staff are not working and VMs may be sitting idle and wants to look at ways to reduce costs without negatively impacting the users when they login each morning. Lastly many users wish to use Microsoft Teams when connected to WVD hosts and use audio and video for their online meetings and although it works, users report regular stuttering during meetings and have asked whether it's possible to improve the experience.

+

+## Description

+

+The goal of this challenge is to demonstrate how you can enable a monitoring solution for WVD and how it may be used to identify and diagnose issues across your deployment. Using this information you should be able to pull in WVD service information, performance metrics and event log information that relates to the service and present it within the WVD insights dashboard. Your environment should also be enabled for automatic scaling based on your companies requirements to optimise costs and ensure sufficient capacity per your peak hour requirements.

+

+**NOTE:** Scaling Automation was deployed in Challenge 9

+

+#### WVD Monitoring

I would make these subsections under 'Description' 3 level headers, not 4 level headers because Description is a 2 level header.

________________________________

In 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618018046&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764662010%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=0ZYu304SPYn0nq72TTfCHSJFGj4mdvLKSRJl4xunDUw%3D&reserved=0>:

> +   - Assume each region starts seeing staff logon between 8am and 9:30am, there is no sudden spike in logons. The office is closed by 7pm.

+   - Users can be forcibly logged off after receiving a 5 minute warning if they are still connected to an active host

+       - Approximately 5% of users in each region may attempt to connect from home each night and WVD should be available to them off-peak across each region and meets the cost saving targets set by the company

+   - WVD Session Hosts should be gradually powered on as morning peak demand increases and existing hosts should be filled up before additional VMs are powered on to reduce excess capacity.

+

+## Success Criteria

+

+To be successful for this challenge you should be able to demonstrate the following;

+

+- Identify the top errors preventing users from connecting over the past 24 hours

+- Select a host pool and show a view with the following information for each WVD host: New Session Status, Stack Version, Current Sessions, Available Sessions

+- Display recent user activity in WVD activity for a specific user including their most recent login and history within the past 30 days

+- Identify performance metrics of a WVD host which has a currently connected user and demonstrate how you can view the "User Input Delay per Process" object and "Max Input Delay" counters for that VM.

+- Enable automatic scaling of WVD session hosts to start and stop machines during peak/off-peak hours across each region. Consider how the peak/off-peak hours and load balancing will influence the configuration. Assume for the purpose of this workshop that VM sizing and user density is designed for a 1CPU per Session to help with testing in a limited size environment.

+

+**Tips**

Tips needs to be a level 2 header

________________________________

In 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618018201&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764662010%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=uKxPlt9wC78RTNFu4qpCD8xBkpGRel9toZjHDNVwOWQ%3D&reserved=0>:

> +- Identify the top errors preventing users from connecting over the past 24 hours

+- Select a host pool and show a view with the following information for each WVD host: New Session Status, Stack Version, Current Sessions, Available Sessions

+- Display recent user activity in WVD activity for a specific user including their most recent login and history within the past 30 days

+- Identify performance metrics of a WVD host which has a currently connected user and demonstrate how you can view the "User Input Delay per Process" object and "Max Input Delay" counters for that VM.

+- Enable automatic scaling of WVD session hosts to start and stop machines during peak/off-peak hours across each region. Consider how the peak/off-peak hours and load balancing will influence the configuration. Assume for the purpose of this workshop that VM sizing and user density is designed for a 1CPU per Session to help with testing in a limited size environment.

+

+**Tips**

+

+> _The logging information may take a few minutes to appear in the monitoring portal. You'll also need to generate user traffic by connecting to WVD after enabling the diagnostics for information to flow through._

+

+> _To simulate connection errors you may wish to attempt logins with invalid credentials, closing the client during the logon attempt and deliberately using a different account between the logon to the WVD portal and the WVD session host which should fail._

+

+> _Simulate an unavailable session host by manually stopping the WVD service "RdAgent"_

+

+

+Advanced Challenges (Optional)

Advanced Challenges needs to be a level 2 header

________________________________

In 037-WindowsVirtualDesktop/Student/10-Monitor-Manage-Performance-Health.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618018595&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764672003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=zwRpaXoSkxfRnz0hF6%2Fm1WeAB5RuLIZdvtQW4JM%2BnSY%3D&reserved=0>:

> +

+To be successful for this challenge you should be able to demonstrate the following;

+

+- Identify the top errors preventing users from connecting over the past 24 hours

+- Select a host pool and show a view with the following information for each WVD host: New Session Status, Stack Version, Current Sessions, Available Sessions

+- Display recent user activity in WVD activity for a specific user including their most recent login and history within the past 30 days

+- Identify performance metrics of a WVD host which has a currently connected user and demonstrate how you can view the "User Input Delay per Process" object and "Max Input Delay" counters for that VM.

+- Enable automatic scaling of WVD session hosts to start and stop machines during peak/off-peak hours across each region. Consider how the peak/off-peak hours and load balancing will influence the configuration. Assume for the purpose of this workshop that VM sizing and user density is designed for a 1CPU per Session to help with testing in a limited size environment.

+

+**Tips**

+

+> _The logging information may take a few minutes to appear in the monitoring portal. You'll also need to generate user traffic by connecting to WVD after enabling the diagnostics for information to flow through._

+

+> _To simulate connection errors you may wish to attempt logins with invalid credentials, closing the client during the logon attempt and deliberately using a different account between the logon to the WVD portal and the WVD session host which should fail._

+

+> _Simulate an unavailable session host by manually stopping the WVD service "RdAgent"_

Please don't use angle brackets on this text. We don't want it to appear as a quote on the rendered page.

________________________________

In 037-WindowsVirtualDesktop/Coach/README.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618019686&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764672003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=79oHuVGTHG2eCw1asGfWYjJze%2F4js40nz8JAquPmR4Q%3D&reserved=0>:

> +Welcome to the Coach's guide for the Windows Virtual Desktop What the Hack. Below are links to each challenge that has specific guidance for each challenge.

+

+## Coach's Guides

+

+- Challenge 0: **[Windows Virtual Desktop  Pre-Reqs](00-Pre-Reqs.md)**

+- Challenge 1: **[Planning your WVD Architecture](01-Plan-WVD-Architecture.md)**

+- Challenge 2: **[Implementing and Managing your WVD Network](02-Implement-Manage-Network.md)**

+- Challenge 3: **[Implement and Manage your storage for FsLogix](03-Implement-Manage-Storage.md)**

+- Challenge 4: **[Create and Manage your Images for your Session Hosts](04-Create-Manage-Images.md)**

+- Challenge 5: **[Create and Configure your WVD HostPools](05-Create-Configure-HostPools.md)**

+- Challenge 6: **[Implement and Manage your FsLogix](06-Implement-Manage-FsLogix.md)**

+- Challenge 7: **[Install and Configure your applications](07-Install-Configure-Apps.md)**

+- Challenge 8: **[Plan and Implement BCDR for your WVD Environment](08-Plan-Implement-BCDR.md)**

+- Challenge 9: **[Automation of your WVD Tasks](09-Automate-WVD-Tasks.md)**

+- Challenge 10: **[Monitor and Manage the performance health of your environment](10-Monitor-Manage-Performance-Health.md)**

+- Challenge 11: **[Configure settings for user experience](11-Configure-User-Experience-Settings.md)**

Make sure this ToC text matches the headers in the Coach guide for each challenge and that same text should be used in the Student Guide.

________________________________

In 037-WindowsVirtualDesktop/Coach/00-Pre-Reqs.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618020034&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764682002%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=NniVyN6GzTGCtSKRW19YeUX0jBbzLQbrQGfJUpd78QE%3D&reserved=0>:

> +## Notes & Guidance

+

+**IMPORTANT:** Check with your stakeholders that you are able to use your current Tenant/Subscription to run this Hack.

+

+Links to install the tooling for this hack:

+

+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)

+- [Azure PowerShell Module](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps?view=azps-5.6.0)

+- [Visual Studio Code](https://code.visualstudio.com/)

+

+If attendees will be using a shared Azure subscription, you should be aware of the following:

+

+- Quotas are set per region.  If you increase the quota in a single region, you need to ensure that all students deploy to the same region.  Or else, they will bump up against the quota limits in the region they deploy to.

+- If there is no access to an administrator who can request quota increases, have the students deploy to different regions to stay within the quota limit of each region.

+

+**NOTE** Remember to point students to learning resources

Add a semicolon after NOTE

________________________________

In 037-WindowsVirtualDesktop/Coach/03-Implement-Manage-Storage.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618020717&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764682002%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=t6sLYX07GaH9npwpKxkNwlYwZYrvPO7EOy0%2FnUm4JO8%3D&reserved=0>:

> @@ -0,0 +1,53 @@

+# Challenge 3: Implement and Manage your storage for FsLogix

+

+[< Previous Challenge](./02-Implement-Manage-Network.md) - **[Home](./README.md)** - [Next Challenge >](./04-Create-Manage-Images.md)

+

+## Notes & Guidance

+

+1. Capacity planning

I would use a bullet here, not a number. You don't have a #2<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F2&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764691995%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=OL2SXN7t%2FQSckjYxKeBM2M7TRfDeQKvOGomR7H2%2B4EA%3D&reserved=0> so it's not really an ordered list.

________________________________

In 037-WindowsVirtualDesktop/Coach/10-Monitor-Manage-Performance-Health.md<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23discussion_r618021960&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764691995%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=3BcO6NYCHld5Jf11lisfhY3K12pR8jCzVsRgLtRYZ0E%3D&reserved=0>:

> +

+1. Modify the WVD Insights portal to display the CPU % performance metric for the current host pool on the front page of the dashboard. You will need to ensure the metric is only displayed on the Overview tab.

+    - Students should be able to show a custom CPU performance metric being added to their edited WVD Insights workbook.

+    - Open the WVD Insights page then "Select Customise"

+    - In the next page "Select Edit"

+    - For faster changes scroll to the bottom and look for the +Add menu  and select "add metric"

+

+    ![Add Metric in Insights Workbook Edit mode](../Images/Challenge10-AddMetricDropdown.png)

+

+    - Metric Scope: Resource Scope

+    - Virtual Machine dropdown: "WVDHostsId" selected and add the VM CPU Performance counter.

+    ![Add CPU Metric](../Images/Challenge10-AddMetricValue.png)

+

+    - Save work by first hitting the done editing button and choosing the "Save As" option before hitting Done edit. Users, if successful will have a new metric counter that they added.

+

+## Reference

Change this to 'Learning Resources', doesn't matter that it's in the coach guide.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FWhatTheHack%2Fpull%2F223%23pullrequestreview-641707015&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764701987%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=g5pTg6JqO9MUuG1WrrRAP0efPrpOPRAcDbyHovtMlzg%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOXQB25HUU2E4X46ZO457JLTJ57CBANCNFSM43B5DX3Q&data=04%7C01%7CJason.Masten%40microsoft.com%7C1acb1c20ae284a347f9508d905320ae2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637546534764711988%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=6qWKqmmNV8T8z9y5iMjnOOK2rNC5olC%2FX9ZRZhT6xIo%3D&reserved=0>.

jrzyshr commented on 037-WindowsVirtualDesktop/Student/Resources/challenge-02_template.json:
```diff
@@ -76,7 +76,7 @@
                 },
                 "templateLink": {
-                    "uri": "https://raw.githubusercontent.com/jamasten/WhatTheHack/main/templates/challenge-02_network.json"
+                    "uri": "https://raw.githubusercontent.com/microsoft/WhatTheHack-1/master/037-WindowsVirtualDesktop/Student/Resources/templates/challenge-02_NetworkTemplate.json"
```
Need to fix this.

WhatTheHack-1 => WhatTheHack

jrzyshr commented on 037-WindowsVirtualDesktop/Student/Resources/challenge-02_template.json:
```diff
@@ -121,7 +121,7 @@
                 },
                 "templateLink": {
-                    "uri": "https://raw.githubusercontent.com/jamasten/WhatTheHack/main/templates/challenge-02_identity.json"
+                    "uri": "https://raw.githubusercontent.com/microsoft/WhatTheHack-1/master/037-WindowsVirtualDesktop/Student/Resources/templates/challenge-02_IdentityTemplate.json"
```
Need to fix this.

WhatTheHack-1 => WhatTheHack

jrzyshr reviewed (changes requested):
Okay... I think we're almost there.

Two more changes requested:
1.) Fix the URL in the Challenge 2 json template.   "WhatTheHack-1" => "WhatTheHack"
2.) Remove all the unpacked ZIP file contents in the /dsc folder.  (sorry!)

jrzyshr reviewed (approved):
Thanks for burning the midnight oil!   We're all good!

gfilicetti reviewed (approved):
Done and done! Brilliant work everyone!!
